pkg/collector/corechecks/snmp/internal/report/report_device_metadata.go

```diff
@@ -328,6 +331,47 @@ func buildNetworkTopologyMetadata(deviceID string, store *metadata.Store) []meta
 	return links
 }
 
+func resolveLocalInterface(deviceID string, interfaceIndexByIDType map[string]map[string]int32, localInterfaceIDType string, localInterfaceID string) (string, string) {
+	if localInterfaceID == "" {
+		return localInterfaceIDType, localInterfaceID
+	}
+
+	var typesToTry []string
+	if localInterfaceIDType == "" {
+		// "smart resolution" by multiple types when localInterfaceIDType is not provided (which is often the case)
+		// CAVEAT: in case the smart resolution returns false positives, the solution is to configure the device to provide a proper localInterfaceIDType
+		typesToTry = []string{"interface_name", "mac_address", "interface_name", "interface_alias", "interface_index"}
+	} else {
+		typesToTry = []string{localInterfaceIDType}
+	}
+	for _, idType := range typesToTry {
+		interfaceIndexByIDValue, ok := interfaceIndexByIDType[idType]
+		if ok {
+			ifIndex, ok := interfaceIndexByIDValue[localInterfaceID]
+			if ok {
+				return metadata.IDTypeNDM, deviceID + ":" + strconv.Itoa(int(ifIndex))
+			}
+		}
+	}
+	return localInterfaceIDType, localInterfaceID
+}
+
+func buildInterfaceIndexByIDType(interfaces []metadata.InterfaceMetadata) map[string]map[string]int32 {
```

Comment:
<!-- Questions are appropriate if you have a potential concern but are not quite sure if it's relevant or not. Asking the author for clarification or investigation can lead to a quick resolution. -->
**❓ question**
For big devices, this could be costly. Have you benchmarked this call ?

Reply:
@pducolin 🤔, hum, even if we have 1000s or 5000s interfaces, I don't see any operation that is has high CPU usage.

I did a quick benchmark just to be sure:

```

func Benchmark_buildInterfaceIndexByIDType_bench(b *testing.B) {
	var interfaces []metadata.InterfaceMetadata
	for i := 0; i < 5000; i++ {
		indexStr := strconv.Itoa(i)
		interfaces = append(interfaces, metadata.InterfaceMetadata{
			DeviceID:   "default:1.2.3.4",
			Index:      int32(i),
			MacAddress: indexStr,
			Name:       "name" + indexStr,
			Alias:      "alias" + indexStr,
		})

	}

	b.ResetTimer() // to avoid counting the setup time
	buildInterfaceIndexByIDType(interfaces)
}

```

Results:
```
pkg: github.com/DataDog/datadog-agent/pkg/collector/corechecks/snmp/internal/report
Benchmark_buildInterfaceIndexByIDType_bench
Benchmark_buildInterfaceIndexByIDType_bench-10    	1000000000	         0.001704 ns/op
PASS
```

If I'm reading that correctly, it takes `0.001704` nanosec per call to `buildInterfaceIndexByIDType(5k interfaces)`

Reply:
Thank you